with-editor.el

```diff
@@ -115,11 +115,16 @@ Determining an Emacsclient executable suitable for the
 current Emacs instance failed.  For more information
 please see https://github.com/magit/magit/wiki/Emacsclient."))))
 
+(defun with-editor--emacsclient-name ()
+  (if (s-starts-with-p (downcase invocation-name) "remacs")
```

Comment:
I don't have a system to test this on, but I was concerned about `Emacs.app`/`Remacs.app` on macOS.